src/code.cloudfoundry.org/loggregator/plumbing/conversion/tov2.go

```diff
@@ -56,19 +56,15 @@ func ToV2(e *events.Envelope, usePreferredTags bool) *v2.Envelope {
 // TODO: Do we still need to do an interface?
 func setV2Tag(e *v2.Envelope, key string, value interface{}, usePreferredTags bool) {
 	if usePreferredTags {
-		switch value.(type) {
-		case string:
-			e.GetTags()[key] = value.(string)
-		case int32, int64:
-			e.GetTags()[key] = fmt.Sprintf("%d")
-		case float64:
-			e.GetTags()[key] = fmt.Sprintf("%f")
-		default:
-			e.GetTags()[key] = fmt.Sprintf("%v", value)
+		if s, ok := value.(string); ok {
+			e.GetTags()[key] = s
+			return
 		}
 
+		e.GetTags()[key] = fmt.Sprintf("%v", value)
```

Comment:
Yea, I am not sure how I feel about this either. It feels like if we do something improperly down the road it could shoot us in the foot, but I would hope that whoever is adding a new tag would in the conversion logic would add a test that would prevent them from passing in a object of some type.